### PR TITLE
dotenv lookup: improve exception handling

### PR DESCRIPTION
**Pull request by nocive**

### CODE REVIEW

File: lookup/dotenv.py [hunk 1]
```diff
@@ -45,6 +45,9 @@ def run(self, terms, variables=None, **kwargs):
             if params['key'] is None:
                 ret.append(data)
             else:
-                ret.append(data[params['key']])
+                try:
+                    ret.append(data[params['key']])
+                except (KeyError) as e:
```
Comment (fho) [inline lookup/dotenv.py:50]:
```suggestion
                except Exception as e:
```

why not wrap all exception that happen?

Comment (nocive) [reply lookup/dotenv.py:47]: cause that's not what i see other plugins do :)
you should handle exceptions that can occur from erroneous plugin usage, AFAIK.

Comment (nocive) [reply lookup/dotenv.py]: changed

Comment (fho) [reply lookup/dotenv.py]: Ok, in the example they also wrap Exception: https://docs.ansible.com/ansible/latest/dev_guide/developing_plugins.html#raising-errors

Btw there are also more specific Exceptions available: https://github.com/ansible/ansible/blob/devel/lib/ansible/errors/__init__.py#L202

**Review by fho — APPROVED** [review]

**Comment by nocive** [comment]
tks @fho

**Pull request merged**